From ac559fad20bbae45332c758abb6a790c3fd341a2 Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
Date: Sun, 29 May 2016 23:34:35 -0700
Subject: [PATCH 3/9] bitmap: Bail out on invalid input to FontFileMakeDir
 instead of calling calloc for 0 bytes

Found by clang static analysis:
    Call to 'calloc' has an allocation size of 0 bytes

Signed-off-by: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
---
 src/bitmap/bitscale.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/bitmap/bitscale.c b/src/bitmap/bitscale.c
index 22747a9..00adcc6 100644
--- a/src/bitmap/bitscale.c
+++ b/src/bitmap/bitscale.c
@@ -1477,6 +1477,10 @@ BitmapScaleBitmaps(FontPtr pf,          /* scaled font */
     lastRow = pfi->lastRow;
 
     nchars = (lastRow - firstRow + 1) * (lastCol - firstCol + 1);
+    if (nchars <= 0) {
+        goto bail;
+    }
+
     glyph = pf->glyph;
     for (i = 0; i < nchars; i++)
     {
-- 
2.9.3

